README.md

```diff
@@ -16,7 +16,9 @@ azure-iot-samples-csharp provides a set of easy-to-understand, continuously-test
 
 ## Prerequisites
 
-- .NET Core SDK 2.1.0 or greater on your development machine.  You can download the .NET Core SDK for multiple platforms from [.NET](https://www.microsoft.com/net/download/all).  You can verify the current version of C# on your development machine using 'dotnet --version'.
+- .NET Core SDK 3.0.0 or greater on your development machine.  You can download the .NET Core SDK for multiple platforms from [.NET](https://www.microsoft.com/net/download/all).  You can verify the current version of C# on your development machine using 'dotnet --version'.
```

Comment:
Please let me know if this causes an issue;  I'd prefer not to set the language version to `preview` if possible, but its an option if needed.

Reply:
@jsquire Do you mean going from .NET Core SDK 2.1 -> 3.0? I'm not sure @jebrando ?

Reply:
Yes;  with Core 3.1 being LTS and the recommended version, I believe it makes sense to bump.  With the track two libraries built on C# 8 concepts, it also helps to avoid using `preview` as the language version or making samples more complicated by manually unrolling asynchronous enumerators.

Reply:
I agree with what you're saying, but don't really have enough personal knowledge to say what we should do here. 

@jebrando @elhorton I think the device SDK still specifies ,NET Core 2.1 [Azure IoT Device SDKs Platform Support](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-device-sdk-platform-support#microsoft-sdks-and-device-platform-support) (Scroll down to **.NET SDK**). I don't know whether the .NET Service SDK similarly specifies .NET Core 2.1 and whether introducing .NET Core 3.1 here is going to be confusing for customers.  Can one of you please help out here?

Reply:
I can't speak to the C# SDK but the Platform Support doc is up to date. Adding @sharmasejal and @prmathur-microsoft to help from the C# side

Reply:
One additional bit of context: this impacts only the recommended SDK for compiling; it does not imply that a new runtime would be needed for deploying.  I've left the target framework for the updated sample as `netcoreapp2.1`.

Apologies for not thinking to call that out earlier.

Reply:
@jsquire That sounds like the right thing to do. Thanks!

Reply:
> One additional bit of context: this impacts only the recommended SDK for compiling; it does not imply that a new runtime would be needed for deploying. I've left the target framework for the updated sample as `netcoreapp2.1`.
> 
> Apologies for not thinking to call that out earlier.

@jsquire I'm not sure I completely understand here. I thought you meant that only compiling the SDK was impacted, but it seems from the note that you added that in order to compile the samples w/o specifying preview you need 3.0, Is that right? And, if so, it seems like the note would also need to be added in the prereqs in the quickstart as well. Please advise

Reply:
Apologies for the confusion.

In order to compile C# code that uses the syntax from v8 (released September 2019), without using the `preview` flag, the .NET Core SDK v3 or higher must be used.   It is possible to use the .NET Core SDK v2.1 to compile if setting the language version to `preview`.  

The Event Hubs client library API was designed around some of the C# 8 concepts, most notably asynchronous enumerators for reading events, which is the scenario covered in the quick start that was updated.  In order to compile the quick start code without modification the .NET Core SDK 3.0+ must be used.   

For developers who wish to use an earlier compiler, they would either need to use the .NET Core SDK 2.1+ and set the language version of the quick start project to `preview` or would need to rewrite the code to manage the enumerable manually, which would look something like the snippet [[here]](https://github.com/Azure/azure-sdk-for-net/issues/9853). 

The distinction that I was trying to make was the difference between which version of the .NET Core SDK was needed to compile the code versus the versions of the .NET Core Runtime or .NET Framework that the output could be used with - meaning which target platforms are supported.  Both the .NET Core SDK v3 and v2.1 can compile such the output can run on [multiple platforms](https://docs.microsoft.com/en-us/dotnet/standard/frameworks) spanning .NET Core, .NET Standard, and the .NET Framework.  

That question is a common concern voiced by developers using the Azure SDKs.   The important take-away here is that the proposed change only impacts folks compiling the quick start (or application) code but does not require that hosting environments be upgraded; any runtime environment that the .NET Core SDK v2.1 supported is also supported by the .NET Core SDK v3 .

Reply:
> And, if so, it seems like the note would also need to be added in the prereqs in the quickstart as well. Please advise

Correct; it appears in the `Prerequisites` section of the README that I added to the quick start.